WIP in progress deployment

fabianofranz opened the pull request:
@spadgett

fabianofranz committed:
WIP in progress deployment

fabianofranz commented:
@spadgett @jwforres

spadgett commented on app/scripts/directives/overviewService.js:
```diff
@@ -48,14 +49,39 @@ angular.module('openshiftConsole')
+        };
+
+        $scope.deploymentInProgress = function(deployments) {
+          return anyDeploymentIsInProgress(deployments);
```
I'd just use the filter directly from the view.

spadgett commented on app/scripts/directives/overviewService.js:
```diff
@@ -48,14 +49,39 @@ angular.module('openshiftConsole')
+          return anyDeploymentIsInProgress(deployments);
+        };
+
+        $scope.isDeploymentLatest = function(deployment) {
```
We have a `deploymentIsLatest` filter you should be able to use.

https://github.com/spadgett/origin-web-console/blob/e32941237fb8e9fbc76a932491f4d56d2455b31e/app/scripts/filters/resources.js#L778

fabianofranz commented:
@spadgett conflicts fixed and addressed comments except for `isDeploymentLatest`: the difference in comparison to the filter is that I don't have the `deploymentConfig` so I traverse the list of dcs to find it.

I'll submit a separate PR to what's missing in the task (mostly handling failure scenarios) and a few improvements I want to do to the code.
